pkg/ingress/model_build_listener.go

```diff
@@ -247,3 +257,199 @@ func (t *defaultModelBuildTask) computeIngressExplicitSSLPolicy(_ context.Contex
 	}
 	return &rawSSLPolicy
 }
+
+type MutualAuthenticationConfig struct {
+	Port                          int64   `json:"port"`
+	Mode                          string  `json:"mode"`
+	TrustStore                    *string `json:"trustStore,omitempty"`
+	IgnoreClientCertificateExpiry *bool   `json:"ignoreClientCertificateExpiry,omitempty"`
+}
+
+func (t *defaultModelBuildTask) computeIngressMutualAuthentication(ctx context.Context, ing *ClassifiedIngress) (map[int64]*elbv2model.MutualAuthenticationAttributes, error) {
+	var rawMtlsConfigStringFromIngressClassParam string
+	var fromIngressClassParams bool
+	var rawMtlsConfigString string
+
+	if ing.IngClassConfig.IngClassParams != nil && ing.IngClassConfig.IngClassParams.Spec.MutualAuthentication != nil {
+		rawMtlsConfigStringFromIngressClassParam = string(*ing.IngClassConfig.IngClassParams.Spec.MutualAuthentication)
+		fromIngressClassParams = true
+	}
+	// If both Ingress and IngressClassParam is missing mutualAuthentication config, return default mutualAuthentication mode
+	if exists := t.annotationParser.ParseStringAnnotation(annotations.IngressSuffixMutualAuthentication, &rawMtlsConfigString, ing.Ing.Annotations); !exists {
+		if !fromIngressClassParams {
+			return map[int64]*elbv2model.MutualAuthenticationAttributes{443: {
+				Mode: string(elbv2model.MutualAuthenticationOffMode),
```

Comment:
What if the users do not have port 443 in the listeners (like they have port 8080), and neither do they specify the mtls setting?

Reply:
also need to call out the default mtls in the ingress annotation and ingress class param doc.

Reply:
This will be ignored then while building the Listener SDK request since we only map the port and their respective setting in mTLS.